tensorflow/python/ops/rnn.py

```diff
@@ -785,7 +785,7 @@ def dynamic_rnn(cell, inputs, sequence_length=None, initial_state=None,
 
   if not time_major:
     # (B,T,D) => (T,B,D)
-    flat_input = tuple(array_ops.transpose(input_, [1, 0, 2])
+    flat_input = tuple(array_ops.transpose(input_, [1, 0]+range(2,input_.get_shape().ndims))
```

Comment:
sadly this won't work if input_ has unknown rank.  you'll have to do something like:

input_rank = input_.get_shape().ndims
if input_rank is None:
  transpose_indices = array_ops.concat(0, ([1, 0], array_ops.range(2, array_ops.rank(input_)))
else:
  transpose_indices = [1, 0] + range(2, input_rank)